app/jsx/pages/BrowsePage.jsx

```diff
@@ -60,6 +64,90 @@ felis tellus mollis orci, sed start a journal nunc eget odio.
       <br/><br/>
     </div>
   )}
+
+}
+
+// Tree traversal help via http://codepen.io/anon/pen/Ftkln
+// 'depts' property (JSON)
+// [ ...,
+//  {"id"=>"international",
+//   "name"=>"UCLA International Institute",
+//   "children"=>
+//    [{"id"=>"asia",
+//      "name"=>"UCLA Asia Institute",
+//      "children"=>[{"id"=>"ccs", "name"=>"UCLA Center for Chinese Studies"}]}, .. ], } .. ]
+class DeptTree extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { depts: this.props.depts }
+    // this.onSelect = this.onSelect.bind(this)
+  }
+  onSelect(node) {
```

Comment:
What's the goal of this onSelect code? It seems like it's trying to deselect the previously selected node, but if that's happening I'm not understanding the effect. Does "select" here mean "expand" in the UI? If so, it seems like this would make it so only one is expanded at a time, which I'm not seeing when I run it.